### PR TITLE
Add controllerTest type as a helper for running controller integration tests

### PR DESCRIPTION
**Pull request by staebler**
Add controllerTest which extracts a lot of the boilerplate for setting up integration tests that test controller logic.

### CODE REVIEW

**Comment by n3wscott** [comment]
WOOO

LGTM

**Comment by kibbles-n-bytes** [comment]
Looks great!

**Comment by MHBauer** [comment]
after the fact, I'm really not happy that I LGTM'd this.

File: test/integration/controller_test.go [hunk 1]
```diff
@@ -1319,3 +512,352 @@ func getTestHappyPathBrokerClientConfig() fakeosb.FakeClientConfiguration {
...
+	ct.catalogClient = catalogClient
+	ct.catalogClientConfig = catalogClientConfig
+	ct.osbClient = osbClient
+	ct.controller = controller
+	ct.informers = informers
+
+	ct.client = catalogClient.ServicecatalogV1beta1()
+
+	if ct.setup != nil {
+		ct.setup(ct)
```
Comment (MHBauer) [inline test/integration/controller_test.go]: This setup function should be called within`newTestController` up above, before starting the controllers.

Comment (staebler) [reply test/integration/controller_test.go]: Would you care to elaborate on why you think that the `setup` function needs to be run before `newTestController`? The types of things that are done in typical `setup` functions use objects that don't even exist until after `newTestController` is run. Is there something being done in the `setup` functions that you think is problematic to run after `newTestController`?

Comment (MHBauer) [reply test/integration/controller_test.go]: It needs to be run inside, not before. It needs to run after some things, but before other things. After the controllers have been created, but before they have been started.

Comment (staebler) [reply test/integration/controller_test.go]: Sorry. I was not clear on what elaboration I was looking for. I was hoping for some details on what the deficiencies are and not on what the proposed solution to the deficiencies are. I believe this is being discussed in other issues and in slack, so feel free to forego providing those details here.

**Pull request merged**